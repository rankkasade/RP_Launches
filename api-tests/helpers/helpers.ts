import {config} from 'dotenv';

config({
    debug: false
});

// 4 minutes timeout for data generation
export const getLongRunningTimeout = () => 5 * 60 * 1000;

export const getProjectName = (): string => process.env.PROJECT_NAME ?? '';

export const getBasePath = (): string => `${process.env.BASE_URL ?? ''}/api`;

export const addConfigToFetch = () => ({
    headers: {
        // Need to be replaced
        Authorization: `Bearer ${process.env.API_KEY ?? ''}`,
    }
});