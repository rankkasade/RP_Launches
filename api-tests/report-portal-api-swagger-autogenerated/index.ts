// tslint:disable
/**
 * Report Portal
 * Report Portal API documentation
 *
 * OpenAPI spec version: 5.11.0
 * Contact: Support Report Portal <support@reportportal.io>
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


export * from "./api";
export * from "./configuration";
