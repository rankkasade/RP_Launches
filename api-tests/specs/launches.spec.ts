import {describe, test} from '@jest/globals';

import {addConfigToFetch, getBasePath, getLongRunningTimeout, getProjectName} from '../helpers/helpers';
import * as ReportPortalAPI from "../report-portal-api-swagger-autogenerated";


describe('Tests for RP Launches API', () => {
    let launchControllerApi: ReportPortalAPI.LaunchControllerApi;
    let demoDataControllerApi: ReportPortalAPI.DemoDataControllerApi;
    let demoDataRs: ReportPortalAPI.DemoDataRs;

    beforeAll(async () => {
        launchControllerApi = new ReportPortalAPI.LaunchControllerApi(undefined, getBasePath());
        demoDataControllerApi = new ReportPortalAPI.DemoDataControllerApi(undefined, getBasePath());

        demoDataRs = await demoDataControllerApi.generateUsingPOST({createDashboard: false}, getProjectName(), addConfigToFetch())
    }, getLongRunningTimeout());

    // GET positive query
    test('should have the right project launches', async () => {
        const data = await launchControllerApi.getAllLaunchNamesUsingGET(getProjectName(), undefined, addConfigToFetch());

        expect(data).toMatchSnapshot();
    });

    // GET negative query
    test('should handle the error in case of accessing to non-existing launch', async () => {
        const neLaunchId: string = 'non-existing-launch-id';
        const responsePromise = launchControllerApi.getLaunchUsingGET(neLaunchId, getProjectName(), addConfigToFetch());

        await expect(responsePromise).rejects.toMatchObject({
            status: 404,
            statusText: "Not Found",
            url: `${getBasePath()}/v1/${getProjectName()}/launch/${neLaunchId}`,
        });
    });

    // POST positive query
    test('should start launch analysis', async () => {
        const data = await launchControllerApi.startLaunchAnalyzerUsingPOST({
            analyzeItemsMode: [
                ReportPortalAPI.AnalyzeLaunchRQ.AnalyzeItemsModeEnum.TOINVESTIGATE
            ],
            analyzerMode: ReportPortalAPI.AnalyzeLaunchRQ.AnalyzerModeEnum.ALL,
            analyzerTypeName: ReportPortalAPI.AnalyzeLaunchRQ.AnalyzerTypeNameEnum.AutoAnalyzer,
            launchId: demoDataRs.launchIds[0]
        }, getProjectName(), addConfigToFetch());

        expect(data).toEqual({
            "message": `autoAnalyzer analysis for launch with ID='${demoDataRs.launchIds[0]}' started.`
        });
    });

    // POST negative query 1
    test('should handle error when analysis started for non-existing launch', async () => {
        const responsePromise = launchControllerApi.startLaunchAnalyzerUsingPOST({
            analyzeItemsMode: [
                ReportPortalAPI.AnalyzeLaunchRQ.AnalyzeItemsModeEnum.TOINVESTIGATE
            ],
            analyzerMode: ReportPortalAPI.AnalyzeLaunchRQ.AnalyzerModeEnum.ALL,
            analyzerTypeName: ReportPortalAPI.AnalyzeLaunchRQ.AnalyzerTypeNameEnum.AutoAnalyzer,
            launchId: 9183829382
        }, getProjectName(), addConfigToFetch());

        await expect(responsePromise).rejects.toMatchObject({
            status: 404,
            statusText: "Not Found",
            url: `${getBasePath()}/v1/${getProjectName()}/launch/analyze`,
        });
    });

    // POST negative query 2
    test('should handle error when bad request is sent', async () => {
        const responsePromise = launchControllerApi.startLaunchAnalyzerUsingPOST({
            analyzeItemsMode: [
                "BAD_PARAM" as unknown as ReportPortalAPI.AnalyzeLaunchRQ.AnalyzeItemsModeEnum
            ],
            analyzerMode: ReportPortalAPI.AnalyzeLaunchRQ.AnalyzerModeEnum.PREVIOUSLAUNCH,
            analyzerTypeName: ReportPortalAPI.AnalyzeLaunchRQ.AnalyzerTypeNameEnum.PatternAnalyzer,
            launchId: demoDataRs.launchIds[0]
        }, getProjectName(), addConfigToFetch());

        await expect(responsePromise).rejects.toMatchObject({
            status: 400,
            statusText: "Bad Request",
            url: `${getBasePath()}/v1/${getProjectName()}/launch/analyze`,
        });
    });

    // PUT positive query
    test('should change launch info', async () => {
        const result = await launchControllerApi.bulkUpdateUsingPUT({
            ids: [demoDataRs.launchIds[0], demoDataRs.launchIds[1]],
            attributes: [],
            description: {
                action: ReportPortalAPI.Description.ActionEnum.CREATE,
                comment: 'Test comment',
            }
        }, getProjectName(), addConfigToFetch());

        expect(result).toMatchSnapshot();
    });

    // PUT negative query 1
    test('should handle error when no launch id is sent', async () => {
        const responsePromise = launchControllerApi.bulkUpdateUsingPUT({
            ids: [],
            attributes: [],
            description: {
                action: ReportPortalAPI.Description.ActionEnum.CREATE,
                comment: 'Test comment',
            }
        }, getProjectName(), addConfigToFetch());

        await expect(responsePromise).rejects.toMatchObject({
            status: 400,
            statusText: "Bad Request",
            url: `${getBasePath()}/v1/${getProjectName()}/launch/info`,
        });
    });

    // PUT negative query 2
    test('should handle error when non existing project name is used', async () => {
        const responsePromise = launchControllerApi.bulkUpdateUsingPUT({
            ids: [1, 2],
            attributes: [],
            description: {
                action: ReportPortalAPI.Description.ActionEnum.CREATE,
                comment: 'Test comment',
            }
        }, 'getProjectName()', addConfigToFetch());

        await expect(responsePromise).rejects.toMatchObject({
            status: 404,
            statusText: "Not Found",
            url: `${getBasePath()}/v1/getProjectName()/launch/info`,
        });
    });

    // DELETE positive query
    test('should delete launch by id', async () => {
        const lastLaunchId = demoDataRs.launchIds[demoDataRs.launchIds.length - 1];
        const data = await launchControllerApi.deleteLaunchUsingDELETE(lastLaunchId, getProjectName(), addConfigToFetch());

        expect(data).toEqual({
            "message": `Launch with ID = '${lastLaunchId}' successfully deleted.`
        });
    });

    // DELETE negative query
    test('should handle error when non-existing launch is deleted', async () => {
        const responsePromise = launchControllerApi.deleteLaunchUsingDELETE(786438764, getProjectName(), addConfigToFetch());

        await expect(responsePromise).rejects.toMatchObject({
            status: 404,
            statusText: "Not Found",
            url: `${getBasePath()}/v1/${getProjectName()}/launch/786438764`,
        });
    });

    afterAll(async () => {
        await launchControllerApi.deleteLaunchesUsingDELETE({ids: demoDataRs.launchIds}, getProjectName(), addConfigToFetch())
    });
});