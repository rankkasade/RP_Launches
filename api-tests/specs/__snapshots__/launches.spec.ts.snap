// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for RP Launches API should change launch info 1`] = `
{
  "message": "Attributes successfully updated",
}
`;

exports[`Tests for RP Launches API should have the right project launches 1`] = `
[
  "Demo Api Tests",
]
`;
